### PR TITLE
ECO Flow Fixes

### LINKED ISSUE #860: Disabling hold timing violations fixing for inputs and outputs
Reported by @shalan 

### Prompt
The ECO flow currently attempts to fix hold timing violations on inputs and outputs, which won't really work if there are timing constraints.

### Proposal
Show how to use the ECO flow to disable hold timing violation fixing for IOs. If the feature doesn't exist, it should be surfaced.

### PR DESCRIPTION
**Pull request by donn**
+ Add option to skip hold timing violation fixing for I/O pins
+ Properly expose ECO flow Tcl commands
~ Formatting changes

---
Resolves #860.

### CODE REVIEW

**Comment by msaligane** [comment]
Thanks @donghl17 and @donn this LGTM! 

@donghl17 I believe you tested this on a few designs (AES, JPEG and Litex mgmt_core)?

**Comment by donghl17** [comment]
@donn I just made an improvement of the eco flow. Could you please give me access to commit to your branch?

**Comment by donn** [comment]
@donghl17 You've recieved access to the branch. Commit what you will. https://github.com/donn/openlane/invitations

**Comment by donn** [comment]
Please be aware I'm not signing off on this unless the CI passes.

**Comment by msaligane** [comment]
@donn I have reviewed with Haolin a few times and this runs through a few design. So I approve it.

**Comment by donn** [comment]
@msaligane GitHub needs to know that too so I can merge :)

**Comment by msaligane** [comment]
Ofcourse! Thanks @donn

**Comment by donn** [comment]
Err, that is... I'm waiting on you. https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/approving-a-pull-request-with-required-reviews

**Comment by donn** [comment]
Alright, well, if you don't care about this PR I see no reason why I should.

**Comment by msaligane** [comment]
@donn I did not see your response...

**Review by msaligane — APPROVED** [review]

File: scripts/tcl_commands/routing.tcl [hunk 1]
```diff
@@ -437,13 +437,13 @@ proc run_routing {args} {
 
     # if diode insertion does *not* happen as part of global routing, then
     # we can insert fill cells early on
-    if { ($::env(DIODE_INSERTION_STRATEGY) != 3) && ($::env(ECO_FINISH) ==1) } {
+    if { ($::env(DIODE_INSERTION_STRATEGY) != 3) && ($::env(ECO_ENABLE) ==0 | $::env(ECO_FINISH) ==1) } {
```
Comment (antonblanchard) [inline scripts/tcl_commands/routing.tcl:455]: Did you mean to use bitwise or here?

Comment (donghl17) [reply scripts/tcl_commands/routing.tcl]: Thanks. The issue has been fixed.

**Comment by donghl17** [comment]
@donn Is there a way to check which part of the code fails the lint test? I just did lint test myself and found there is one error of "unable to import odb" and 40 conventions. I am wondering if the odb error occurs because I am not using the right pylint config, or I am supposed to solve all the conventions to pass the test.

**Comment by donn** [comment]
We don't do pylint. You can see what linters we use here: https://github.com/The-OpenROAD-Project/OpenLane/runs/5222722298?check_suite_focus=true

**Review by donghl17 — DISMISSED** [review]

File: scripts/gen_insert_buffer.py [hunk 2]
```diff
@@ -81,7 +82,7 @@
             vio_name = drcSections[i].strip()
             report_end_str = re.search("min_report_end", vio_name)
             if report_end_str is not None:
-                print("LLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLL")
+                print("not complete report")
```
Comment (msaligane) [inline scripts/gen_insert_buffer.py:85]: @donghl17 Can you change this to something like: report is in incomplete

**Review by msaligane — CHANGES_REQUESTED** [review]

File: scripts/gen_insert_buffer.py (hunk 2 above)
Comment (antonblanchard) [reply scripts/gen_insert_buffer.py]: I noticed this message when testing. If we hit here, doesn't it mean we collected all the records and it is complete?

Comment (antonblanchard) [reply scripts/gen_insert_buffer.py]: Longer term, I also wonder if there is a better to parse format we can use, vs the somewhat brittle parsing of the text output.

Comment (msaligane) [reply scripts/gen_insert_buffer.py]: I think this is totally possible. The reports are generated directly out OpenROAD/OpenSTA, so we can have a command that does this without having to parse the reports (which are most likely subject to change). This should be up to the flow owners/maintainers to coordinate.

**Review by msaligane — APPROVED** [review]

**Pull request merged**